aws-cpp-sdk-access-management/CMakeLists.txt

```diff
@@ -33,7 +33,7 @@ if(USE_WINDOWS_DLL_SEMANTICS AND BUILD_SHARED_LIBS)
     add_definitions("-DAWS_ACCESS_MANAGEMENT_EXPORTS")
 endif()
 
-add_library(${PROJECT_NAME} ${ACCESS_MANAGEMENT_SRC})
+add_library(${PROJECT_NAME} SHARED ${ACCESS_MANAGEMENT_SRC})
```

Comment:
> always build dynamic libraries

why always build shared libraries?

Reply:
to simplify the required code changes, as well as the implementation here: https://github.com/ceph/ceph/pull/38673/files#diff-34eff6689abb9572612fd8b57b58beefc7fa86fd2bb54efd27dfe628f67bfa62

Reply:
i don't really follow on how shared libraries simplifies the required code change. could you be more specific? the change `ceph.spec.in` does increase the amount of code we need to maintain, not to mention the potential conflicts with libaws
 installed by user.

Reply:
ok. now i understand, supporting both shared and static libs (based on come cmake conf) is more complex.
supporting only static libs will simplify :-)
is this acceptable solution to statically link the sdk libs to the rgw binaries?

Reply:
@tchaikov @yuvalif we discussed this in RGW's upstream standup;  by far the consensus is we would strongly prefer to use dynamic linking of the (very large) SDK.  Among the rasons, 1) if static linking is used, the SDK will be incorporated into at least 3 different objects (radosgw, librgw, radosgw-admin);  2) it is common practice to run multiple instances of radosgw on a single container/vm--this is a classic case were dynamic linking is more efficient at runtime,

Reply:
@tchaikov are you ok with keeping the shared libs solution?